backend/src/routes/weaviate.ts

```diff
@@ -1,126 +1,95 @@
-    .do();
+    .do()
+}
 
-  // const { userInput } = req.body;
-
-  res.status(201).json({ response: JSON.stringify(result, null, 2) });
-});
+weaviateRouter.get('/', async (_req, res) => {
+  const generatePrompt =
+    'You are an assistant. Answer ONLY with the facts given as a context. Answer shortly for the questions given by the user. Question: {What is the role of DevOps in software development?}'
+  const result = await executeWeaviateQuery(generatePrompt, ['devops'])
+  res.status(200).json({ response: JSON.stringify(result, null, 2) })
+})
 
-weaviateRouter.get("/test", async (_req, res) => {
+weaviateRouter.get('/test', async (_req, res) => {
   const result = await client.graphql
     .get()
-    .withClassName("ResearchPaper")
-    .withHybrid({
-      query: "software",
-    })
+    .withClassName('ResearchPaper')
+    .withHybrid({ query: 'software' })
     .withLimit(3)
-    .withFields("title content")
-    .do();
-
-  res.status(201).json({ response: JSON.stringify(result, null, 2) });
-});
-
-weaviateRouter.post("/search", async (req, res) => {
-  const inputString = req.body.text;
+    .withFields('title content')
+    .do()
+  res.status(200).json({ response: JSON.stringify(result, null, 2) })
+})
 
+weaviateRouter.post('/search', async (req, res) => {
+  const inputString = req.body.text
   if (!inputString) {
-    res.status(400).json({ error: "text is required in request body" });
+    return res.status(400).json({ error: 'text is required in request body' })
   }
 
-  const promptText = `You are an entity extractor now. Extract entities from the following text: '${inputString}'`;
+  const promptText = `You are an entity extractor now. Extract entities from the following text: '${inputString}'`
   try {
     const response = await axios.post(
       OPEN_AI_API_URL,
       {
-        model: "gpt-3.5-turbo",
-        messages: [{ role: "system", content: promptText }],
+        model: 'gpt-3.5-turbo',
+        messages: [{ role: 'system', content: promptText }],
         temperature: 0.7,
       },
       {
         headers: {
           Authorization: `Bearer ${OPEN_AI_APIKEY}`,
-          "Content-Type": "application/json",
+          'Content-Type': 'application/json',
         },
       }
-    );
-    const entities = response.data.choices[0].message.content.trim();
-    const generatePrompt = `You are an assistant. Answer ONLY with the facts given as a context. Answer shortly for the questions given by the user. Question: ${inputString}`;
-
-    const result = await client.graphql
-      .get()
-      .withClassName("ResearchPaper")
-      .withGenerate({
-        groupedTask: generatePrompt,
-        groupedProperties: ["content", "title"],
-      })
-      .withNearText({
-        concepts: [entities],
-      })
-      .withFields("chunk_index")
-      .withLimit(4)
-      .do();
+    )
 
-    console.log(JSON.stringify(result, null, 2));
+    const entities = response.data.choices[0].message.content.trim()
```

Comment:
Nitpick:
To get a bit more robustness, I would consider checking nulls here:
`response?.data?.choices[0]?.message?.content.trim()` etc

OpenAI's API not the most reliable service

Reply:
Good attention, now resloved...